utils/apiObjects.ts

```diff
@@ -0,0 +1,151 @@
+import { hexToDecimal, stringToFelt } from "./feltService";
+import { formatHexString } from "./stringService";
+
+const STARKNET = formatHexString(stringToFelt("starknet"));
+const DISCORD = formatHexString(stringToFelt("discord"));
+const TWITTER = formatHexString(stringToFelt("twitter"));
+const GITHUB = formatHexString(stringToFelt("github"));
+const PROOF_OF_PERSONHOOD = formatHexString(
+  stringToFelt("proof_of_personhood")
+);
+
+export class Identity {
+  private data: IdentityData;
+  private userDataMap: Map<string, string>;
+  private verifierDataMap: Map<string, string>;
+
+  constructor(identityData: IdentityData) {
+    this.data = identityData;
+
+    this.userDataMap = new Map();
+    identityData.user_data.forEach((ud) => {
+      this.userDataMap.set(ud.field, ud.data);
+    });
+
+    this.verifierDataMap = new Map();
+    identityData.verifier_data.forEach((vd) => {
+      this.verifierDataMap.set(`${vd.verifier}_${vd.field}`, vd.data);
+    });
+  }
+
+  getData(): IdentityData {
+    return this.data;
+  }
+
+  getUserData(field: string): string | undefined {
+    return this.userDataMap.get(field);
+  }
+
+  getVerifierData(verifier: string, field: string): string | undefined {
+    return this.verifierDataMap.get(`${verifier}_${field}`);
+  }
+
+  getDomain(): string | undefined {
+    return this.data.domain?.domain;
+  }
+
+  getDomainExpiry(): number | undefined {
+    return this.data.domain?.expiry;
+  }
+
+  isMain(): boolean {
```

Comment:
should be tested :)

Reply:
Alright alright, see: https://github.com/starknet-id/app.starknet.id/pull/580/commits/4c049f26559f822acea2beedad03e66b34821bf9